code/api/src/tests/modules/user/mutations.test.js

```diff
@@ -0,0 +1,149 @@
+      .expect(200)
+    expect(updateResponse.body.data.userUpdate.streetAddress1).toEqual('123 New Lane')
+    expect(updateResponse.body.data.userUpdate.streetAddress1).not.toEqual(null)
+    expect(updateResponse.body.data.userUpdate.city).toEqual('Denver')
+    expect(updateResponse.body.data.userUpdate.city).not.toEqual(null)
+    expect(updateResponse.body.data.userUpdate.state).toEqual('CO')
+    expect(updateResponse.body.data.userUpdate.state).not.toEqual(null)
+    expect(updateResponse.body.data.userUpdate.zip).toEqual('80207')
+    expect(updateResponse.body.data.userUpdate.zip).not.toEqual(null)
+  })
+
+  it('can update a user with description', async () => {
+    const tokenResponse = await request(server)
+      .get('/')
+      .send({query: '{ userLogin(email: "user@crate.com", password: "123456", role: "user") { token user { id } } }'})
+    const userID = tokenResponse.body.data.userLogin.user.id
+
+    const updateResponse = await request(server)
+      .post('/')
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send({ query: `mutation { userUpdate(id: ${userID}, description: "The only user") { id email streetAddress1 streetAddress2 city state zip description image } }` })
+      .expect(200)
+    expect(updateResponse.body.data.userUpdate.description).toEqual('The only user')
+  })
+
+  it('can update a user with image', async () => {
+    const tokenResponse = await request(server)
+      .get('/')
+      .send({query: '{ userLogin(email: "user@crate.com", password: "123456", role: "user") { token user { id } } }'})
+    const userID = tokenResponse.body.data.userLogin.user.id
+
+    const updateResponse = await request(server)
+      .post('/')
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send({ query: `mutation { userUpdate(id: ${userID}, image: "image.jpg") { id email streetAddress1 streetAddress2 city state zip description image } }` })
+      .expect(200)
+    expect(updateResponse.body.data.userUpdate.image).toEqual('image.jpg')
+  })
+
+  it('can update a user delivery date', async () => {
+    const tokenResponse = await request(server)
+      .get('/')
+      .send({query: '{ userLogin(email: "user@crate.com", password: "123456", role: "user") { token user { id } } }'})
+    const userID = tokenResponse.body.data.userLogin.user.id
+
+    const updateResponse = await request(server)
+      .post('/')
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+      .send({ query: `mutation { userUpdate(id: ${userID}, deliveryDate: "15th" ) { id email streetAddress1 streetAddress2 city state zip description image deliveryDate } }` })
+      .expect(200)
+    expect(updateResponse.body.data.userUpdate.deliveryDate).toEqual('15th')
+  })
```

Comment:
Great job with these thorough tests.

Reply:
Thank you for getting these done so quickly.